frontend/src/libs/components/input/input.tsx

```diff
@@ -5,36 +5,71 @@ import {
   type FieldValues,
 } from 'react-hook-form';
 
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
 import { useFormController } from '~/libs/hooks/hooks.js';
 
+import styles from './styles.module.scss';
+
 type Properties<T extends FieldValues> = {
   control: Control<T, null>;
   errors: FieldErrors<T>;
-  label: string;
+  label?: string;
   name: FieldPath<T>;
   placeholder?: string;
   type?: 'text' | 'email' | 'password';
+  isActive?: boolean;
+  isFilled?: boolean;
+  isDisabled?: boolean;
 };
 
 const Input = <T extends FieldValues>({
   control,
   errors,
-  label,
+  label = '',
   name,
   placeholder = '',
   type = 'text',
+  isActive,
+  isFilled,
+  isDisabled,
 }: Properties<T>): JSX.Element => {
   const { field } = useFormController({ name, control });
 
   const error = errors[name]?.message;
   const hasError = Boolean(error);
+  const inputStyles = [
+    styles.input,
+    isActive && styles.active,
+    isFilled && styles.filled,
+    isDisabled && styles.disabled,
+    error && styles.error,
+  ];
 
-  return (
-    <label>
+  const InputContent = (
+    <>
       <span>{label}</span>
-      <input {...field} type={type} placeholder={placeholder} />
-      {hasError && <span>{error as string}</span>}
-    </label>
+      <div className={styles.inputWrapper}>
+        <input
+          {...field}
+          type={type}
+          placeholder={placeholder}
+          className={getValidClassNames(...inputStyles)}
+          disabled={isDisabled}
+        />
+        {type === 'password' && (
+          <span className={styles.passwordEye}>&#128065;</span>
+        )}
+      </div>
+      {hasError && (
+        <span className={styles.errorMessage}>{error as string}</span>
+      )}
+    </>
+  );
+
+  return label ? (
```

Comment:
What is the purpose that? 

Just return InputContent JSX wrapped by div
Also, there is no need to create InputContent const

Reply:
fixed